test/scenario/metering-plans/src/test/test.js

```diff
@@ -0,0 +1,156 @@
+      end: timestamp,
+      organization_id: orgID,
+      dedup_id: uuid.v4(),
+      space_id: 'sampler-space-id',
+      consumer_id: 'sampler-consumer-id',
+      resource_id: resourceId,
+      plan_id: planName,
+      resource_instance_id: 'sampler-resource-instance-id',
+      measured_usage: [{
+        measure: 'duration',
+        quantity: hourInMillis * hours
+      }]
+    };
+
+    return usageDoc;
+  };
+
+  const sendUsage = async (usage) => {
+    const resp = await doPost(':url/v1/metering/collected/usage', {
+      url: env.collectorURL,
+      headers: authHeader(systemToken),
+      body: usage
+    });
+
+    expect(resp.statusCode).to.equal(202);
+    return resp.headers.location;
+  };
+
+  const createMeteringMapping = async(resourceType, planName, meteringPlanID, token) =>
+    await doPost(':url/v1/provisioning/mappings/metering/resources/:resource_type/plans/:plan_name/:plan_id', {
+      url: env.provisioningURL,
+      headers: authHeader(token),
+      resource_type: resourceType,
+      plan_name: planName,
+      plan_id: meteringPlanID
+    });
+
+  const createRatingMapping = async (resourceType, planName, meteringPlanID, token) =>
+    await doPost(':url/v1/provisioning/mappings/rating/resources/:resource_type/plans/:plan_name/:plan_id', {
+      url: env.provisioningURL,
+      headers: authHeader(token),
+      resource_type: resourceType,
+      plan_name: planName,
+      plan_id: meteringPlanID
+    });
+
+  const createPricingMapping = async (resourceType, planName, meteringPlanID, token) =>
+    await doPost(':url/v1/provisioning/mappings/pricing/resources/:resource_type/plans/:plan_name/:plan_id', {
+      url: env.provisioningURL,
+      headers: authHeader(token),
+      resource_type: resourceType,
+      plan_name: planName,
+      plan_id: meteringPlanID
+    });
+
+  before(async() => {
+    if(env.secured) {
+      systemToken = oauth.cache(env.authServerURL, env.systemClientId, env.systemClientSecret,
+        'abacus.usage.read abacus.usage.write'
+      );
+
+      const promisifiedTokenStart = util.promisify(systemToken.start);
+      await promisifiedTokenStart();
+    }
+
+    setEventuallyPollingInterval(env.pollInterval);
+
+    await Promise.all([
+      createMeteringMapping(resourceId, planName, meteringPlanId, systemToken),
+      createRatingMapping(resourceId, planName, meteringPlanId, systemToken),
+      createPricingMapping(resourceId, planName, meteringPlanId, systemToken)
+    ]);
+  });
+
+  context('submit usages', () => {
+    beforeEach(() => {
+      orgID = `sampler-plan-scenario-${uuid.v4()}`;
+    });
+
+    it('generated report should be correct', async() => {
+      const usageHours = {
+        firstDoc: 2,
+        secondDoc: 4,
+        thirdDoc: -1
+      };
+      const currentMonth = 0;
+      const monthsReport = 4;
+      const usageHoursIndex = 0;
+      const samplerTestResourceIdIndex = 0;
+
+      usageDoc = buildUsageDoc(orgID, usageHours.firstDoc);
```

Comment:
Don't mutate a single variable. Either inline the method calls or use names like `firstUsage`, `secondUsage`, etc.